Cmdline/Action/Replace.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using log4net;
+
+namespace CKAN.CmdLine
+{
+    public class Replace : ICommand
+    {
+        private static readonly ILog log = LogManager.GetLogger(typeof(Replace));
+
+        public IUser User { get; set; }
+
+        public Replace(IUser user)
+        {
+            User = user;
+        }
+
+
+        public int RunCommand(CKAN.KSP ksp, object raw_options)
+        {
+            ReplaceOptions options = (ReplaceOptions) raw_options;
+
+            if (options.ckan_file != null)
+            {                
+                options.modules.Add(LoadCkanFromFile(ksp, options.ckan_file).identifier);
+            }
+
+            if (options.modules.Count == 0 && ! options.replace_all)
+            {
+                // What? No mods specified?
+                User.RaiseMessage("Usage: ckan replace Mod [Mod2, ...]");
+                User.RaiseMessage("  or   ckan replace --all");
+                return Exit.BADOPT;
+            }
+
+            User.RaiseMessage("\r\nReplacing modules...\r\n");
+            var registry = RegistryManager.Instance(ksp).registry;
+            var to_replace = new List<ModuleReplacement>();
+
+            if (options.replace_all)
+            {
+                log.Debug("Running Replace all");
+                var installed = new Dictionary<string, Version>(registry.Installed());
```

Comment:
`Installed` already returns `Dictionary<string, Version>`. Do we really need to make a copy of it? That will presumably cost CPU time. This should work:
```
var installed = registry.Installed();
```

Reply:
I'm not sure why it is like that. I copied the code from the Upgrade command, and that bit is unchanged

Reply:
Fair enough; it would make sense to reserve this for a separate project that addresses all commands at once, if needed.